cpp/oneapi/dal/test/engine/linalg/loops.hpp

```diff
@@ -0,0 +1,69 @@
+/*******************************************************************************
+* Copyright 2020-2021 Intel Corporation
+*
+* Licensed under the Apache License, Version 2.0 (the "License");
+* you may not use this file except in compliance with the License.
+* You may obtain a copy of the License at
+*
+*     http://www.apache.org/licenses/LICENSE-2.0
+*
+* Unless required by applicable law or agreed to in writing, software
+* distributed under the License is distributed on an "AS IS" BASIS,
+* WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+* See the License for the specific language governing permissions and
+* limitations under the License.
+*******************************************************************************/
+
+#pragma once
+
+#include "oneapi/dal/test/engine/linalg/matrix.hpp"
+
+namespace oneapi::dal::test::engine::linalg {
+
+template <typename T, layout lyt, typename Op>
+inline void enumerate_row_first(const matrix<T, lyt>& m, Op&& op) {
```

Comment:
Do we need to have such helpers using sycl kernels in the future?

Reply:
This is part of test engine, don't think we'll need to compute anything on GPU in tests. 

Initially I thought that this code will be reused for DAL kernels and tests, but, in fact, we have different requirements for these functionality. For testing, this must be as simple as possible, but for DAL kernels deliver high performance.